src/Model/List/ListDecorator.php

```diff
@@ -5,39 +5,36 @@
 use SilverStripe\Model\ModelData;
 use LogicException;
 use Traversable;
+use BadMethodCallException;
 
 /**
  * A base class for decorators that wrap around a list to provide additional
  * functionality. It passes through list methods to the underlying list
  * implementation.
  *
- * @template TList of SS_List&Sortable&Filterable&Limitable
  * @template T
  * @implements SS_List<T>
- * @implements Sortable<T>
- * @implements Filterable<T>
- * @implements Limitable<T>
  */
-abstract class ListDecorator extends ModelData implements SS_List, Sortable, Filterable, Limitable
+abstract class ListDecorator extends ModelData implements SS_List
 {
     /**
-     * @var TList<T>
+     * @var SS_List<T>
```

Comment:
Revert this and any other reference to `TList`

Reply:
Reverted